source/com.microsoft.tfs.client.common/src/com/microsoft/tfs/client/common/credentials/CredentialsHelper.java

```diff
@@ -161,4 +174,141 @@ public static TFSConnection getVstsRootConnection(final TFSConnection connection
 
         return vstsConnection;
     }
+
+    public static Credentials getOAuthCredentials(final URI serverURI) {
+        removeStaleOAuth2Token();
+
+        final Authenticator authenticator;
+        final Token token;
+
+        if (serverURI != null) {
+            log.debug("Interactively retrieving credential based on oauth2 flow for " + serverURI.toString()); //$NON-NLS-1$
+            log.debug("Trying to persist credential, generating a PAT"); //$NON-NLS-1$
+
+            authenticator = new VstsPatAuthenticator(CLIENT_ID, REDIRECT_URL, accessTokenStore, tokenStore);
+
+            final String tokenKey =
+                authenticator.getUriToKeyConversion().convert(serverURI, authenticator.getAuthType());
+            removeStalePersonalAccessToken(tokenKey, serverURI);
+
+            token = authenticator.getPersonalAccessToken(
+                serverURI,
+                VsoTokenScope.AllScopes,
+                PatCredentials.USERNAME_FOR_CODE_ACCESS_PAT,
```

Comment:
This is the display name for the PAT on VSTS, do you want to use some other string rather than the hardcoded username?